web_scraping.py

```
@@ -0,0 +1,92 @@
+        print(f"Found headers: {headers}")
+        
+        for row in table.find_all('tr')[1:]:
+            cells = row.find_all(['td', 'th'])
+            if len(cells) != len(headers):
+                print("Skipping row due to mismatch in columns.")
+                continue
+            row_data = [cell.get_text(strip=True) for cell in cells]
+            data.append(row_data)
+        
+        print(f"Parsed {len(data)} rows.")
+        return headers, data
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)